linera-service/src/cli_wrappers/local_kubernetes_net.rs

```diff
@@ -89,19 +85,12 @@ impl LineraNetConfig for LocalKubernetesNetConfig {
             self.testing_prng_seed,
             self.binaries_dir,
             KubectlInstance::new(Vec::new()),
-            future::ready(
-                (0..self.num_initial_validators)
-                    .map(|_| async {
-                        KindCluster::create()
-                            .await
-                            .expect("Creating kind cluster should not fail")
-                    })
-                    .collect::<Vec<_>>(),
-            )
-            .then(join_all)
-            .await
-            .into_iter()
-            .collect::<Vec<_>>(),
+            future::join_all((0..self.num_initial_validators).map(|_| async {
+                KindCluster::create()
+                    .await
+                    .expect("Creating kind cluster should not fail")
+            }))
+            .await,
```

Comment:
I would personally execute the (very imperative) expression before with `let clusters = future::join_all(..);`